addons/hr_timesheet/models/project.py

```diff
@@ -108,12 +108,16 @@ def _search_is_project_overtime(self, operator, value):
             raise ValueError(_('Invalid operator: %s') % operator)
 
         query = """
-            SELECT P.id
-              FROM project_project P
-         LEFT JOIN project_task T ON P.id = T.project_id
-             WHERE T.planned_hours IS NOT NULL
-          GROUP BY P.id
-            HAVING SUM(T.remaining_hours) < 0
+            SELECT Project.id
+            FROM project_project AS Project
+                     LEFT JOIN project_task AS Task ON Project.id = Task.project_id
+                     LEFT JOIN project_task_type AS Stage ON Task.stage_id = Stage.id
+            WHERE Task.planned_hours IS NOT NULL
+              AND Project.allow_timesheets = TRUE
+              AND Task.parent_id IS NULL
+              AND (Stage.fold IS NULL OR Task.stage_id IS NULL)
+            GROUP BY Project.id
+            HAVING SUM(Task.remaining_hours) < 0
```

Comment:
I am wondering if we could not use `is_closed` column, I think `is_closed` is stored and indexed in task model and so we could use it to avoid doing an extra left join. 🙂 

What do you think? 🙂 

```suggestion
            SELECT Project.id
              FROM project_project AS Project
              JOIN project_task AS Task 
                ON Project.id = Task.project_id
             WHERE Task.planned_hours IS NOT NULL
               AND Project.allow_timesheets = TRUE
               AND Task.parent_id IS NULL
               AND Task.is_closed IS NULL
          GROUP BY Project.id
            HAVING SUM(Task.remaining_hours) < 0
```

Also, I think we could directly do a JOIN instead of `LEFT JOIN` since `Task.planned_hours IS NOT NULL` will excluded the project without any task, right? 🤔 

(I formatted the sql as we did in upgrade script) 🙂 😇

Reply:
Good point, I always forget that there is a stored related for is_closed in tasks, this simplifies the query.
When it comes to the formatting, I am not a fan of this one in general, but I know that usually chs is annoying everyone in the company about it, trying to convert us all. xD